.github/workflows/pytest.yml

```diff
@@ -17,20 +17,32 @@ jobs:
     runs-on: ubuntu-latest
     steps:
       - name: Setup Ocean.py
-        uses: actions/checkout@v2
+        uses: actions/checkout@v3
       - name: Set up Python 3.8
         uses: actions/setup-python@v2
         with:
           python-version: '3.8'
-      - uses: actions/checkout@v2
+      - uses: actions/checkout@v3
         name: Checkout Barge
         with:
           repository: "oceanprotocol/barge"
           path: 'barge'
+      - name: Set ADDRESS_FILE
+        run: echo "ADDRESS_FILE=${HOME}/.ocean/ocean-contracts/artifacts/address.json" >> $GITHUB_ENV
+      - name: Login to Docker Hub
+        if: ${{ env.DOCKERHUB_PASSWORD && env.DOCKERHUB_USERNAME }}
+        run: |
+          echo "Login to Docker Hub";echo "$DOCKERHUB_PASSWORD" | docker login -u "$DOCKERHUB_USERNAME" --password-stdin
+        env:
+          DOCKERHUB_USERNAME: ${{ secrets.DOCKERHUB_USERNAME }}
+          DOCKERHUB_PASSWORD: ${{ secrets.DOCKERHUB_PASSWORD }}
       - name: Run Barge
         working-directory: ${{ github.workspace }}/barge
         run: |
           bash -x start_ocean.sh --no-dashboard 2>&1 --with-provider2 --with-c2d --with-thegraph --skip-subgraph-deploy > start_ocean.log &
+        env:
+          CONTRACTS_VERSION: predictoor
```

Comment:
I'm thinking more about this. Needing a special version just for predictoor -- just to work around immunefi policy -- is gonna be a huge pain.

I created a new github issue to discuss this more, and get to a resolution. [predictoor-pm#18](https://github.com/oceanprotocol/predictoor-pm/issues/18)

Reply:
it's not just immunify.   I will release a new version of contracts as soon as we have the entire flow tested, with UI.    Otherwise, we will release a new patch every couple of hours :)